### ISSUE TITLE
keyboard init error

### ISSUE THREAD

**Issue by Josscii** [moses-palmer/pynput#21]
I just use following code to init a keyboard:
```python
from pynput.keyboard import Key, Controller
keyboard = Controller()
```
but end up with errors in console:
```
File "/usr/local/lib/python3.6/site-packages/pynput/keyboard/_darwin.py", line 134, in __init__
    self._mapping = get_unicode_to_keycode_map()
  File "/usr/local/lib/python3.6/site-packages/pynput/_util/darwin.py", line 163, in get_unicode_to_keycode_map
    with keycode_context() as context:
  File "/usr/local/Cellar/python3/3.6.0/Frameworks/Python.framework/Versions/3.6/lib/python3.6/contextlib.py", line 82, in __enter__
    return next(self.gen)
  File "/usr/local/lib/python3.6/site-packages/pynput/_util/darwin.py", line 128, in keycode_context
    layout_data = layout.bytes().tobytes()
AttributeError: 'NoneType' object has no attribute 'bytes'
```
Why this happened?

**Comment by moses-palmer** [comment]
It appears that `TISCopyCurrentKeyboardInputSource` has returned `NULL`. Under what circumstances do you use *pynput*? Do you run it in a restricted context or over SSH?

**Comment by moses-palmer** [comment]
Perhaps you could add some debug logging around [here](https://github.com/moses-palmer/pynput/blob/master/lib/pynput/_util/darwin.py#L124) to check `keyboard` and `layout`?

**Comment by Josscii** [comment]
I just use pynput on my own Mac, I try to check the `layout`'s value here, and it turns out to be None. So I just add a Not-None Checking in your code here, it works but now keyboard can only recognize the `Ctrl` or `alt`, the normal key like `a` is not work.

**Comment by Josscii** [comment]
I'm thinking that is there anything to do with the input source? I do install a third-party input source.

**Comment by moses-palmer** [comment]
I think you're right. There should be a check against `None` return, but other than that I don't think there is much to do if the keyboard layout cannot be retrieved.

**Comment by Josscii** [comment]
Okay, maybe it's just my mac's wired problem, I've tried it on other mac, and there's no problem to recognize the keyboard. Thanks for your help and awesome library, it helps me a lot.
And One more problem happened while I'm trying to move mouse and then click the left button, but it same that it's just too quick to do the two things at the same time, so the click event not work at times, so I have to delay some seconds to make it work, code are here:
```
m.position = (arr[0], arr[1])
time.sleep(1) # if i don't sleep here, the click may be not working
m.click(Button.left)
m.click(Button.left)
```

**Comment by moses-palmer** [comment]
Could you please test [fixup-mac-unknown-keyboard-layout](https://github.com/moses-palmer/pynput/tree/fixup-mac-unknown-keyboard-layout)?

I don't have access to a Mac at the moment, so I had to do some blind coding. The fixup branch is supposed to prevent the `AttributeError`; since the layout is unknown, you will probably still get other errors, but the goal is to get at least a numeric keycode in the end.

**Comment by Josscii** [comment]
I'd like to do it. Should I just update `pynput` with pip?

**Comment by moses-palmer** [comment]
You'll have to clone the repository, and then run your script with `PYTHONPATH="<path to repo>/lib" python ...`.

**Comment by Josscii** [comment]
![image](https://cloud.githubusercontent.com/assets/6898115/24848359/f97221d2-1df8-11e7-8ea7-c0f761cd05e6.png)
is this line that you modified? I think I can just change to same code as your fix, if it works, I will post here to tell you.

**Comment by moses-palmer** [comment]
That is correct.

**Comment by Josscii** [comment]
![image](https://cloud.githubusercontent.com/assets/6898115/24848584/009a2fa8-1dfa-11e7-8852-044a696eb970.png)
It seems that it has a syntax error.
![image](https://cloud.githubusercontent.com/assets/6898115/24848608/2673572c-1dfa-11e7-9dc1-e2a9eaa2caf8.png)

**Comment by moses-palmer** [comment]
Perhaps you could copy `darwin.py` from [here](https://github.com/moses-palmer/pynput/blob/fixup-mac-unknown-keyboard-layout/lib/pynput/_util/darwin.py) and replace your local copy?

**Comment by Josscii** [comment]
![image](https://cloud.githubusercontent.com/assets/6898115/24849144/e549d066-1dfc-11e7-8a7c-98b263eaa415.png)
new bugs....

**Comment by moses-palmer** [comment]
Hmm, these line numbers do not match the latest master. Do you think that you could run it from the repository?

**Comment by moses-palmer** [comment]
I have released *pynput 1.3* which fixes this issue.

The fix prevents the crash, but some functionality may not be available.
